### PR TITLE
call out to arcgis-rest-groups for get, create, update, search, remove, protect, etc.

### LINKED ISSUE #117: use existing @esri/arcgis-rest-groups for group search, CRUD, and protect methods
Most, if not all of these fns https://github.com/Esri/ember-arcgis-portal-services#groups-service can be achieved via https://esri.github.io/arcgis-rest-js/api/groups/

I've opened this here (instead of the Hub backlog) since it is purely a shuffling of code refactor and would not result in deduping code in the Hub (yet).

### PR DESCRIPTION
**Pull request by jgravois**
resolves #117 

threw @tomwayson a bone toward #129 as well.

full disclosure. there were zero unit tests here for groups to begin with and the dummy app only has UI elements to exercise ~75% of the affected functions.

if you want me to npm link the branch and run e2e, just say the word.

### CODE REVIEW

File: tests/unit/mixins/service-mixin-test.js [hunk 1]
```diff
@@ -119,6 +119,11 @@ test('addOptions, authenticated session with authenticated portal options', func
     authMgr: { portal: 'https://session.host.com' },
   });
 
+  // no clue
+  // stub(subject, 'getPortalRestUrl', function (obj) {
```
Comment (tomwayson) [inline tests/unit/mixins/service-mixin-test.js]: When you do this:
https://github.com/Esri/ember-arcgis-portal-services/blob/be7539547501f1e63c13ac1c4267fb4f9de83459/tests/unit/services/hosted-service-test.js#L3-L4

You'll get `this.stub()` in tests:

https://github.com/Esri/ember-arcgis-portal-services/blob/be7539547501f1e63c13ac1c4267fb4f9de83459/tests/unit/services/hosted-service-test.js#L20-L23

I think case, I don't think you'd even need to `resolve()`, you'd just return the expected URL.

File: package-lock.json [hunk 2]
```diff
@@ -39,45 +39,54 @@
...
     "@esri/arcgis-rest-common-types": {
-      "version": "1.7.1",
-      "resolved": "https://registry.npmjs.org/@esri/arcgis-rest-common-types/-/arcgis-rest-common-types-1.7.1.tgz",
-      "integrity": "sha512-b38kmPx9yI13Ksi/R6TdDHWBQJbmU2MfveVWtJgL4PaeK6bBDKcbPaKbdXrWZbzoGVtWX+teL4B61TS/IQB4eg==",
-      "dev": true
+      "version": "1.11.1",
+      "resolved": "https://registry.npmjs.org/@esri/arcgis-rest-common-types/-/arcgis-rest-common-types-1.11.1.tgz",
+      "integrity": "sha512-0fCQmiLf3zDQjuEprd0514tMyiWiVVMfGrXF9omEgnfinsn1Cjyt9N2IX3yrhszE1xaUVP+5ieYcsAJ+4G8Bcg=="
+    },
+    "@esri/arcgis-rest-groups": {
```
[the remark this answers is not in the document]
Comment (jgravois) [reply package-lock.json]: i can't say i know why, but groups is _already_ an EAPS dependency

https://github.com/Esri/ember-arcgis-portal-services/blob/345d702c02b749f6a1d32ed7c6819666d1ac3f1c/package.json#L33

[the remark this answers is not in the document]
Comment (tomwayson) [reply package-lock.json]: Ah, of course, b/c of the notifications code that Jupe added.

File: tests/unit/mixins/service-mixin-test.js (hunk 1 above)
Comment (jgravois) [reply tests/unit/mixins/service-mixin-test.js]: swapping in the new `imports` broke ~10 tests. i'm gonna leave the icing off the 🍰 for now.

File: addon/services/groups-service.js [hunk 3]
```diff
@@ -54,24 +68,32 @@ export default Service.extend(serviceMixin, {
...
   },
 
   /**
    * Update an existing group
    */
   update (group, portalOpts) {
-    const urlPath = `/community/groups/${group.id}/update?f=json`;
-    return this._post(urlPath, group, portalOpts);
+    const args = this.addOptions({ group }, portalOpts);
+    return updateGroup(args)
```
[the remark this answers is not in the document]
Comment (jgravois) [reply addon/services/groups-service.js]: absolutely. addressed this (and the rest of your good feedback) in 983ff9b

**Pull request merged**